src/Compilers/Core/Portable/Compilation/SymbolInfo.cs

```diff
@@ -10,51 +10,33 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    public struct SymbolInfo : IEquatable<SymbolInfo>
+    public readonly struct SymbolInfo : IEquatable<SymbolInfo>
     {
-        internal static readonly SymbolInfo None = new SymbolInfo(null, ImmutableArray<ISymbol>.Empty, CandidateReason.None);
+        internal static readonly SymbolInfo None = default;
 
-        private ImmutableArray<ISymbol> _candidateSymbols;
+        private readonly ImmutableArray<ISymbol> _candidateSymbols;
 
         /// <summary>
-        /// The symbol that was referred to by the syntax node, if any. Returns null if the given
-        /// expression did not bind successfully to a single symbol. If null is returned, it may
-        /// still be that case that we have one or more "best guesses" as to what symbol was
-        /// intended. These best guesses are available via the CandidateSymbols property.
+        /// The symbol that was referred to by the syntax node, if any. Returns null if the given expression did not
+        /// bind successfully to a single symbol. If null is returned, it may still be that case that we have one or
+        /// more "best guesses" as to what symbol was intended. These best guesses are available via the <see
+        /// cref="CandidateSymbols"/> property.
         /// </summary>
         public ISymbol? Symbol { get; }
 
         /// <summary>
-        /// If the expression did not successfully resolve to a symbol, but there were one or more
-        /// symbols that may have been considered but discarded, this property returns those
-        /// symbols. The reason that the symbols did not successfully resolve to a symbol are
-        /// available in the CandidateReason property. For example, if the symbol was inaccessible,
-        /// ambiguous, or used in the wrong context.
+        /// If the expression did not successfully resolve to a symbol, but there were one or more symbols that may have
+        /// been considered but discarded, this property returns those symbols. The reason that the symbols did not
+        /// successfully resolve to a symbol are available in the <see cref="CandidateReason"/> property. For example,
+        /// if the symbol was inaccessible, ambiguous, or used in the wrong context.
         /// </summary>
-        public ImmutableArray<ISymbol> CandidateSymbols
-        {
-            get
-            {
-                return _candidateSymbols.NullToEmpty();
-            }
-        }
-
-        internal ImmutableArray<ISymbol> GetAllSymbols()
-        {
-            if (this.Symbol != null)
-            {
-                return ImmutableArray.Create<ISymbol>(this.Symbol);
-            }
-            else
-            {
-                return _candidateSymbols;
-            }
-        }
+        /// <remarks>Will never return a <see langword="default"/> array.</remarks>
+        public ImmutableArray<ISymbol> CandidateSymbols => _candidateSymbols.NullToEmpty();
```

Comment:
Externally and internally everything goes through .CandidateSymbols, which means everything always sees a non-default array.  This includes if you have the `default/None` instances of this type.